src/allmydata/web/root.py

```diff
@@ -182,9 +183,44 @@ def data_import_path(self, ctx, data):
     def render_my_nodeid(self, ctx, data):
         tubid_s = "TubID: "+self.client.get_long_tubid()
         return T.td(title=tubid_s)[self.client.get_long_nodeid()]
+
     def data_my_nickname(self, ctx, data):
         return self.client.nickname
 
+    def renderHTTP(self, ctx):
+        req = inevow.IRequest(ctx)
+        t = get_arg(req, "t")
+        if t == "json":
+            return self.json_welcome(ctx)
+        return rend.Page.renderHTTP(self, ctx)
+
+    def json_welcome(self, ctx):
+        inevow.IRequest(ctx).setHeader("content-type", "text/plain")
```

Comment:
I just checked, and my browser (FF52) displays all of `text/plain`, `text/json`, and `application/json` as plain text, so any of them are good for me (I'm pretty sure that `application/json` used to force a download, but apparently it doesn't now). Let's use `application/json` since that's what Github's API uses. We should probably add a `; charset=utf-8` too.